specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,1026 @@
+supported downlevel OS SKUs.
+
+**Toast Composition**: There are well known gaps in the Windows SDK today in the area of composing
+the Toast content. The Windows Toolkit gets around this limitation by providing a well-defined
+object model to compose toasts but all developers don't use the Toolkit. We will provide well
+defined Toast templates as part of our API constructs with a plan to build a Windows Toolkit like
+ToastBuilder model in the near future.
+
+# Description
+
+At a high level, we need to provide a way for all Win32 applications to display toasts irrespective
+of their app type. This includes unpackaged apps and packaged win32 (MSIX Desktop Bridge, MSIX
+Win32App, Sparse Signed Packages). Moreover, all Toast scenarios should adhere to OS resource
+management policies like Power Saver, Network Attribution (amount of data an app uses), enterprise
+group policies, etc. The Windows App SDK Push component will abstract away the complexities of
+dealing with Toast delivery and toast related activations as much as possible freeing the developer
+to focus on other app related challenges.
+
+We will prioritize the following feature set for Toast Notifications in Windows App SDK:
+
+-   Toast Registrations for Unpackaged apps.
+-   Toast Registrations for Packaged apps
+-   Cloud sourced Toasts for Unpackaged Win32 apps.
+-   Cloud sourced Toasts for Packaged Win32 apps.
+-   Local Toast delivery and Create/Update/Delete support for Packaged Win32 apps.
+-   Local Toasts delivery and Create/Update/Delete support for unpackaged Win32 apps.
+
+The following feature set will not be prioritized for the future include:
+
+-   Object Model to compose Toasts without having to rely on Xml templates.
+-   Support for toast collections.
+
+# Examples
+
+## Packaged App Toast Registration
+
+This scenario is specifically geared towards Packaged apps that need to send a toast. An app
+Developer Registers their app for Toast Notifications and sets it up for background\foreground
+activation. Adding a ToastActivated event handler signals that the app is ready to accept and handle
+an action in the foreground where the user clicks on a Toast in the action centre or a UI element in
+the toast body (like a Button). The developer can inturn run some business logic in their app in
+response to the Toast Action.
+
+```cpp
+int main()
+{
+    auto activationInfo = ToastActivationInfo::CreateFromActivationGuid(winrt::guid(c_comServerId));
+    ToastNotificationManager::Default().RegisterActivator(activationInfo);
+
+    auto args = AppInstance::GetCurrent().GetActivatedEventArgs();
+    auto kind = args.Kind();
+
+    if (kind == ExtendedActivationKind::Launch)
+    {
+        // App is launched in Foreground. So intercept toast activators via Foreground event
+        const auto token = ToastNotificationManager::Default().ToastActivated([](const auto&, ToastActivatedEventArgs const& toastActivatedEventArgs)
```

Comment:
this line (and others) uses both east and west const - choose one, be constistent [sic] ;)

Reply:
Fair enough!